pallets/phala/src/compute/wrapped_balances.rs

```diff
@@ -133,6 +144,226 @@ pub mod pallet {
 		ReferendumOngoing,
 		/// The Iteration exceed the max limitaion
 		IterationsIsNotVaild,
+
+		LockLargerThanTotalBalance,
+
+		LiquidityRestrictions,
+	}
+
+	impl<T: Config> Currency<T::AccountId> for Pallet<T>
+	where
+		BalanceOf<T>: sp_runtime::traits::AtLeast32BitUnsigned + Copy + FixedPointConvert + Display,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + pallet_balances::Config + vault::Config,
+		T: pallet_balances::Config<Balance = BalanceOf<T>>,
+	{
+		type Balance = BalanceOf<T>;
+		type PositiveImbalance = PositiveImbalanceOf<T>;
+		type NegativeImbalance = NegativeImbalanceOf<T>;
+		fn total_balance(who: &T::AccountId) -> Self::Balance {
```

Comment:
Total balance and free balance are wrong. By definition:

- Total Balance = Transferrable + Locked + Reserved
- Free = Transferrable + Locked

The total value should reflect the full balance of the account. In the wrapped balances context, it should additionally represent the share value as well. After all, all the shares are also voting power. It's not counted here.

The free balance is exactly the total voting power. It includes the locked balance. So in this context, it should be the share value without excluding the locks.

I believe if you write a proper unit test, you can easily figure out the problems in your code.